timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.kt

```diff
@@ -0,0 +1,808 @@
+      explanation = "Since Timber handles String.format automatically, you may not use String#format().",
+      category = MESSAGES,
+      priority = 5,
+      severity = WARNING,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_THROWABLE = Issue.create(
+      id = "ThrowableNotAtBeginning",
+      briefDescription = "Exception in Timber not at the beginning",
+      explanation = "In Timber you have to pass a Throwable at the beginning of the call.",
+      category = MESSAGES,
+      priority = 5,
+      severity = WARNING,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_BINARY = Issue.create(
+      id = "BinaryOperationInTimber",
+      briefDescription = "Use String#format()",
+      explanation = "Since Timber handles String#format() automatically, use this instead of String concatenation.",
+      category = MESSAGES,
+      priority = 5,
+      severity = WARNING,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_ARG_COUNT = Issue.create(
+      id = "TimberArgCount",
+      briefDescription = "Formatting argument types incomplete or inconsistent",
+      explanation = "When a formatted string takes arguments, you need to pass at least that amount of arguments to the formatting call.",
+      category = MESSAGES,
+      priority = 9,
+      severity = ERROR,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_ARG_TYPES = Issue.create(
+      id = "TimberArgTypes",
+      briefDescription = "Formatting string doesn't match passed arguments",
+      explanation = "The argument types that you specified in your formatting string does not match the types of the arguments that you passed to your formatting call.",
+      category = MESSAGES,
+      priority = 9,
+      severity = ERROR,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_TAG_LENGTH = Issue.create(
+      id = "TimberTagLength",
+      briefDescription = "Too Long Log Tags",
+      explanation = "Log tags are only allowed to be at most" + " 23 tag characters long.",
+      category = CORRECTNESS,
+      priority = 5,
+      severity = ERROR,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+    val ISSUE_EXCEPTION_LOGGING = Issue.create(
+      id = "TimberExceptionLogging",
+      briefDescription = "Exception Logging",
+      explanation = "Explicitly including the exception message is redundant when supplying an exception to log.",
+      category = CORRECTNESS,
+      priority = 3,
+      severity = WARNING,
+      implementation = Implementation(WrongTimberUsageDetector::class.java, JAVA_FILE_SCOPE)
+    )
+
+    val issues = arrayOf(
```

Comment:
Might as well use `listOf`

Reply:
I tried this, but then had to convert back for lint tests, iirc.

Reply:
Actually nvm, some tests are passing all the issues, instead of the individual issues they're testing against.  By simplifying that, I can clean this up too.